cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -204,36 +203,59 @@ def store_flow_cell_data_in_housekeeper(self, flow_cell: FlowCellDirectoryData)
 
     def add_sample_fastq_files_to_housekeeper(self, flow_cell: FlowCellDirectoryData) -> None:
         """Add sample fastq files from flow cell to Housekeeper."""
-        valid_sample_fastq_paths = get_valid_sample_fastq_paths(flow_cell.path)
 
-        for sample_fastq_path in valid_sample_fastq_paths:
-            if self.fastq_path_should_be_stored_in_housekeeper(
-                sample_fastq_path=sample_fastq_path,
-                sequencer_type=flow_cell.sequencer_type,
-                flow_cell_name=flow_cell.id,
-            ):
-                self.store_fastq_path_in_housekeeper(
-                    sample_fastq_path=sample_fastq_path, flow_cell_name=flow_cell.id
+        sample_internal_ids: List[str] = get_sample_internal_ids_from_sample_sheet(flow_cell)
+
+        for sample_internal_id in sample_internal_ids:
+            self.add_bundle_and_version_if_non_existent(sample_internal_id)
+
+            sample_fastqs: Optional[List[Path]] = get_sample_fastqs_from_flow_cell(
+                flow_cell_directory=flow_cell.path, sample_internal_id=sample_internal_id
+            )
+
+            if not sample_fastqs:
+                LOG.warning(
+                    f"Cannot find fastq files for sample {sample_internal_id} in {flow_cell.path}. Skipping."
                 )
+                continue
 
-    def store_fastq_path_in_housekeeper(self, sample_fastq_path: Path, flow_cell_name: str) -> None:
-        sample_id = get_sample_id_from_sample_fastq(sample_fastq_path)
+            for sample_fastq_path in sample_fastqs:
```

Comment:
```suggestion
            for sample_fastq_path in sample_fastq_paths:
```
Okay, I can understand that. For the sake of consistency regarding paths.

Reply:
Fixed in https://github.com/Clinical-Genomics/cg/pull/2237/commits/e0ae47442b03398436b5504bb196a75825c537ee